src/sentry/workflow_engine/migrations/0014_aci_model_updates_from_milestones.py

```diff
@@ -0,0 +1,175 @@
+# Generated by Django 5.1.1 on 2024-11-20 23:39
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import sentry.db.models.fields.foreignkey
+import sentry.db.models.fields.hybrid_cloud_foreign_key
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    dependencies = [
+        ("sentry", "0792_add_unique_index_apiauthorization"),
+        ("workflow_engine", "0013_related_name_conditions_on_dcg"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "workflow_engine_detector" ADD COLUMN "config" boolean NOT NULL DEFAULT true;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "workflow_engine_detector" DROP COLUMN "config";
+                    """,
+                    hints={"tables": ["workflow_engine_detector"]},
+                ),
+            ],
+            state_operations=[
+                migrations.AddField(
+                    model_name="detector",
+                    name="config",
+                    field=models.JSONField(blank=True, default=dict, null=True),
```

Comment:
Why does this have `null=True` and the sql has `NOT NULL`?

I also updated our migration framework, so you can actually skip this whole sql step: https://develop.sentry.dev/api-server/application-domains/database-migrations/#adding-columns-with-a-default

tl;dr: `models.JSONField(db_default=dict)`